src/createServer.js

```diff
@@ -1,13 +1,208 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const expenses = [];
+  const users = [];
+  let expenseIdCounter = 1;
+  let userIdCounter = 1;
+  const app = express();
+
+  app.use(express.json());
+
+  app.get('/expenses', (req, res) => {
+    const { userId, category, from, to, categories } = req.query;
+    let filteredExpenses = expenses;
+
+    if (userId) {
+      filteredExpenses = filteredExpenses.filter(
+        (exp) => exp.userId === parseInt(userId)
```

Comment:
Consider using `Number(userId)` instead of `parseInt(userId)` to handle cases where `userId` might not be a valid number, as `parseInt` can return a number even if the input is not fully numeric.